contracts/authorities/BondingCurveAuthority.sol

```diff
@@ -0,0 +1,223 @@
+        uint80 supply;
+        uint16 creatorFee;
+    }
+
+    constructor(
+        address payable partyDao,
+        uint16 initialPartyDaoFeeBps,
+        uint16 initialTreasuryFeeBps
+    ) {
+        partyDaoFeeBps = initialPartyDaoFeeBps;
+        treasuryFeeBps = initialTreasuryFeeBps;
+        PARTY_DAO = partyDao;
+    }
+
+    function createParty(
+        BondingCurvePartyOptions memory partyOpts
+    ) external payable returns (Party party) {
+        if (partyOpts.creatorFee > 500) {
+            revert InvalidCreatorFee();
+        }
+
+        address[] memory authorities = new address[](1);
+        authorities[0] = address(this);
+
+        party = partyOpts.partyFactory.createParty(
+            partyOpts.partyImpl,
+            authorities,
+            partyOpts.opts,
+            new IERC721[](0),
+            new uint256[](0),
+            0
+        );
+
+        partyInfos[party] = PartyInfo({
+            creator: payable(msg.sender),
+            supply: 0,
+            creatorFee: partyOpts.creatorFee
+        });
+
+        buyPartyCards(party, 1);
+    }
+
+    function createPartyWithMetadata(
+        BondingCurvePartyOptions memory partyOpts,
+        MetadataProvider customMetadataProvider,
+        bytes memory customMetadata
+    ) external payable returns (Party party) {
+        if (partyOpts.creatorFee > 500) {
+            revert InvalidCreatorFee();
+        }
+
+        address[] memory authorities = new address[](1);
+        authorities[0] = address(this);
+
+        party = partyOpts.partyFactory.createPartyWithMetadata(
+            partyOpts.partyImpl,
+            authorities,
+            partyOpts.opts,
+            new IERC721[](0),
+            new uint256[](0),
+            0,
+            customMetadataProvider,
+            customMetadata
+        );
+
+        partyInfos[party] = PartyInfo({
+            creator: payable(msg.sender),
+            supply: 0,
+            creatorFee: partyOpts.creatorFee
+        });
+
+        buyPartyCards(party, 1);
+    }
+
+    function buyPartyCards(Party party, uint80 amount) public payable {
+        PartyInfo memory partyInfo = partyInfos[party];
+
+        if (partyInfo.creator == address(0)) {
+            revert PartyNotSupported();
+        }
+
+        uint256 bondingCurvePrice = _getBondingCurvePrice(partyInfo.supply, amount);
+        uint256 partyDaoFee = (bondingCurvePrice * partyDaoFeeBps) / BPS;
+        uint256 treasuryFee = (bondingCurvePrice * treasuryFeeBps) / BPS;
+        uint256 creatorFee = (bondingCurvePrice * partyInfo.creatorFee) / BPS;
+
+        if (
+            amount == 0 || msg.value != bondingCurvePrice + partyDaoFee + treasuryFee + creatorFee
+        ) {
+            revert InvalidMessageValue();
+        }
+
+        partyInfos[party].supply = partyInfo.supply + amount;
+
+        payable(address(party)).transfer(treasuryFee);
+        if (creatorFee != 0) partyInfo.creator.transfer(creatorFee);
```

Comment:
When called in `createParty()` this will transfer the creator's fees right back to them. Could be optimized.